src/service/engine/index.ts

```diff
@@ -56,4 +89,52 @@ export class EngineService {
 
     return engines;
   }
+
+  async create(
+    name: string,
+    region: string | undefined = undefined,
+    engine_type: string | EngineType = EngineType.GENERAL_PURPOSE,
+    spec: string | undefined = undefined,
+    scale: number | undefined = undefined,
+    auto_stop: number | undefined = undefined,
+    warmup: string | WarmupMethod | undefined = undefined,
+    fail_if_exists: boolean = true
```

Comment:
to many arguments it is better to have first argument `name: string` and second all optional params in separate object.